kokoro/gcp_ubuntu/cmake/build.sh

```diff
@@ -40,7 +40,7 @@ echo "Initializing submodules"
 # so a build failure only prevents building/testing things that depend on it and
 # we can still run the other tests.
 echo "Building with cmake"
-./build_tools/cmake/build.sh
+./build_tools/cmake/clean_build.sh
```

Comment:
Relating to my comment above, you could comment here on why we don't want to use caching (or why it matters / doesn't matter).

Reply:
I think which script the CI uses can be a separate decision (PR) than splitting the scripts. For now this maintains the current behavior.

Reply:
SGTM